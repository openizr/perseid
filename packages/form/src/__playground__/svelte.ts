/* c8 ignore start */
/* eslint-disable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call */

import { SvelteComponent } from 'svelte';
import Engine from 'scripts/core/Engine';
import Form from 'scripts/svelte/Form.svelte';
import Field from '__playground__/Field.svelte';

let app: Form;
const { log } = console;

function main(): void {
  app = new Form({
    props: {
      Field: Field as typeof SvelteComponent | null | undefined,
      configuration: {
        root: 'root',
        fields: {
          test: { type: 'string', required: true },
        },
        steps: {
          root: {
            fields: ['test'],
          },
        },
        plugins: [
          (engine: Engine): void => {
            engine.on('userAction', (data, next) => {
              log(data);
              return next(data);
            });
          },
        ],
      },
    },
    target: document.getElementById('root') as unknown as HTMLElement,
  });
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Vue components tree.
window.addEventListener('beforeunload', () => {
  (app as { $destroy: () => void; }).$destroy();
});
