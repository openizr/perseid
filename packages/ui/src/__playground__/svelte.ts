/* c8 ignore start */

import type { SvelteComponent } from 'svelte';
import Router from '__playground__/pages/Router.svelte';

let app: SvelteComponent;

function main(): void {
  const target = document.querySelector('#root') as unknown as HTMLElement;
  target.innerHTML = '';
  app = new Router({
    hydrate: false,
    target,
  });
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Svelte components tree.
window.addEventListener('beforeunload', () => {
  app.$destroy();
});
