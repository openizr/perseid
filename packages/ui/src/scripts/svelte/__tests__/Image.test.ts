/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import { render } from '@testing-library/svelte';
import UIImage from 'scripts/svelte/Image.svelte';

describe('svelte/UIImage', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const { container } = render(UIImage, {
      props: {
        alt: 'test', ratio: 'square', src: 'https://test.com/a.jpg', modifiers: 'large',
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - no lazy', () => {
    const { container } = render(UIImage, {
      props: {
        alt: 'test', ratio: 'square', src: 'https://test.com/a.jpg', lazy: false,
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - custom ratio', () => {
    const { container } = render(UIImage, { props: { alt: 'test', ratio: '25x32', src: 'https://test.com/a.jpg' } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - no lazy custom ratio', () => {
    const { container } = render(UIImage, {
      props: {
        alt: 'test', lazy: false, ratio: '25x32', src: 'https://test.com/a.jpg',
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    const { container } = render(UIImage, {
      props: {
        id: 'test', alt: 'test', ratio: 'portrait', src: 'https://test.com/a.jpg',
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - standard ratio with itemprop', () => {
    const { container } = render(UIImage, {
      props: {
        itemProp: 'image', alt: 'test', ratio: 'landscape', src: 'https://test.com/a.jpg',
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - custom ratio with itemprop', () => {
    const { container } = render(UIImage, {
      props: {
        itemProp: 'image', alt: 'test', ratio: '1x5', src: 'https://test.com/a.jpg',
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with panoramic ratio', () => {
    const { container } = render(UIImage, { props: { alt: 'test', ratio: 'panoramic', src: 'https://test.com/a.jpg' } });
    expect(container.firstChild).toMatchSnapshot();
  });
});
