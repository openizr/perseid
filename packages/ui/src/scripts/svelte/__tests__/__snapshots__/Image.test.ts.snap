// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`svelte/UIImage > renders correctly - basic 1`] = `
<div>
  <div
    class="ui-image ui-image--square--large"
  >
    <img
      alt="test"
      height="1"
      loading="lazy"
      src="https://test.com/a.jpg"
      width="1"
    />
  </div>
  
  <!--&lt;Image&gt;-->
</div>
`;

exports[`svelte/UIImage > renders correctly - custom ratio 1`] = `
<div>
  <img
    alt="test"
    class="ui-image ui-image--25x32"
    height="32"
    loading="lazy"
    src="https://test.com/a.jpg"
    width="25"
  />
  
  <!--&lt;Image&gt;-->
</div>
`;

exports[`svelte/UIImage > renders correctly - custom ratio with itemprop 1`] = `
<div>
  <img
    alt="test"
    class="ui-image ui-image--1x5"
    height="5"
    itemprop="image"
    loading="lazy"
    src="https://test.com/a.jpg"
    width="1"
  />
  
  <!--&lt;Image&gt;-->
</div>
`;

exports[`svelte/UIImage > renders correctly - no lazy 1`] = `
<div>
  <div
    class="ui-image ui-image--square"
  >
    <img
      alt="test"
      height="1"
      src="https://test.com/a.jpg"
      width="1"
    />
  </div>
  
  <!--&lt;Image&gt;-->
</div>
`;

exports[`svelte/UIImage > renders correctly - no lazy custom ratio 1`] = `
<div>
  <img
    alt="test"
    class="ui-image ui-image--25x32"
    height="32"
    src="https://test.com/a.jpg"
    width="25"
  />
  
  <!--&lt;Image&gt;-->
</div>
`;

exports[`svelte/UIImage > renders correctly - standard ratio with itemprop 1`] = `
<div>
  <div
    class="ui-image ui-image--landscape"
  >
    <img
      alt="test"
      height="2"
      itemprop="image"
      loading="lazy"
      src="https://test.com/a.jpg"
      width="3"
    />
  </div>
  
  <!--&lt;Image&gt;-->
</div>
`;

exports[`svelte/UIImage > renders correctly - with id 1`] = `
<div>
  <div
    class="ui-image ui-image--portrait"
    id="test"
  >
    <img
      alt="test"
      height="3"
      loading="lazy"
      src="https://test.com/a.jpg"
      width="2"
    />
  </div>
  
  <!--&lt;Image&gt;-->
</div>
`;

exports[`svelte/UIImage > renders correctly - with panoramic ratio 1`] = `
<div>
  <div
    class="ui-image ui-image--panoramic"
  >
    <img
      alt="test"
      height="9"
      loading="lazy"
      src="https://test.com/a.jpg"
      width="16"
    />
  </div>
  
  <!--&lt;Image&gt;-->
</div>
`;
